godump.go

```diff
@@ -67,35 +72,113 @@ func htmlColorize(code, str string) string {
 	return fmt.Sprintf(`<span style="color:%s">%s</span>`, htmlColorMap[code], str)
 }
 
+// Dumper holds configuration for dumping structured data.
+// It controls depth, item count, and string length limits.
+type Dumper struct {
+	maxDepth     int
+	maxItems     int
+	maxStringLen int
+}
+
+// Option defines a functional option for configuring a Dumper.
+type Option func(*Dumper) *Dumper
+
+// WithMaxDepth allows to control how deep the structure will be dumped.
+// Param n must be 0 or greater or this will be ignored, and default MaxDepth will be 15
+func WithMaxDepth(n int) Option {
+	return func(d *Dumper) *Dumper {
+		if n >= 0 {
+			d.maxDepth = n
+		}
+		return d
+	}
+}
+
+// WithMaxItems allows to control how much items from an array, slice or maps can be printed.
+// Param n must be 0 or greater or this will be ignored, and default MaxItems will be 100
+func WithMaxItems(n int) Option {
+	return func(d *Dumper) *Dumper {
+		if n >= 0 {
+			d.maxItems = n
+		}
+		return d
+	}
+}
+
+// WithMaxStringLen allows to control how long can printed strings be.
+// Param n must be 0 or greater or this will be ignored, and default MaxStringLen will be 100000
+func WithMaxStringLen(n int) Option {
+	return func(d *Dumper) *Dumper {
+		if n >= 0 {
+			d.maxStringLen = n
+		}
+		return d
+	}
+}
+
+// NewDumper creates a new Dumper with the given options applied.
+// Defaults are used for any setting not overridden.
+func NewDumper(opts ...Option) *Dumper {
+	d := &Dumper{
+		maxDepth:     defaultMaxDepth,
+		maxItems:     defaultMaxItems,
+		maxStringLen: defaultMaxStringLen,
+	}
+	for _, opt := range opts {
+		d = opt(d)
+	}
+	return d
+}
+
 // Dump prints the values to stdout with colorized output.
 func Dump(vs ...any) {
+	defaultDumper.Dump(vs...)
+}
+
+// Dump prints the values to stdout with colorized output.
+func (d *Dumper) Dump(vs ...any) {
 	printDumpHeader(os.Stdout, 3)
 	tw := tabwriter.NewWriter(os.Stdout, 0, 0, 1, ' ', 0)
-	writeDump(tw, vs...)
+	d.writeDump(tw, vs...)
 	tw.Flush()
 }
 
 // Fdump writes the formatted dump of values to the given io.Writer.
 func Fdump(w io.Writer, vs ...any) {
+	defaultDumper.Fdump(w, vs...)
+}
+
+// Fdump writes the formatted dump of values to the given io.Writer.
+func (d *Dumper) Fdump(w io.Writer, vs ...any) {
 	printDumpHeader(w, 3)
 	tw := tabwriter.NewWriter(w, 0, 0, 1, ' ', 0)
-	writeDump(tw, vs...)
+	d.writeDump(tw, vs...)
 	tw.Flush()
 }
 
 // DumpStr dumps the values as a string with colorized output.
 func DumpStr(vs ...any) string {
+	return defaultDumper.DumpStr(vs...)
+}
+
+// DumpStr dumps the values as a string with colorized output.
+func (d *Dumper) DumpStr(vs ...any) string {
```

Comment:
I would call this one Format

Reply:
Similar to another thread I'd ideally keep the naming similar to the original API. "Format" sounds generic and doesn't specifically tell me what it formats or how it formats.

DumpStr implies more of what it does, it returns a string

I've updated the comments for both of these too

```go
// DumpStr returns a string representation of the values with colorized output.
```